client/image_import.go

```diff
@@ -42,5 +45,7 @@ func (cli *Client) ImageImport(ctx context.Context, source ImageImportSource, re
 	if err != nil {
 		return ImageImportResult{}, err
 	}
-	return ImageImportResult{rc: resp.Body}, nil
+	return ImageImportResult{
+		Body: newCancelReadCloser(ctx, resp.Body),
+	}, nil
```

Comment:
This exports the underlying read closer; so that it's possible to mock these ... using less tacky solutions than I currently have in the CLI; 🙈 🙈 🙈 🙈 
https://github.com/docker/cli/blob/606f1c65d09eb77821ca9ca7ffb42bd5302328da/cli/command/image/load_test.go#L77


```go
func mockImageLoadResult(content string) client.ImageLoadResult {
	out := client.ImageLoadResult{}

	// Set unexported field "body"
	v := reflect.ValueOf(&out).Elem()
	f := v.FieldByName("body")
	r := io.NopCloser(strings.NewReader(content))
	reflect.NewAt(f.Type(), unsafe.Pointer(f.UnsafeAddr())).Elem().Set(reflect.ValueOf(r))
	return out
}
```

Reply:
Unfortunately that doesn't work outside of the client package itself, because the `body` (or `rc`) field is not exported, and the method's signature requires a concrete type (`ImageLoadResult`) not an interface;


<img width="771" height="277" alt="Screenshot 2025-11-03 at 15 46 50" src="https://github.com/user-attachments/assets/37569d02-7955-46f0-bbc8-ef4839ae6eac" />

Reply:
No I meant embed the io.ReadCloser, so you still have access to it but you don't need to specify the fieldname when using `io.ReadAll`.

```go
	body := io.NopCloser(strings.NewReader(""))
	resp := client.ImageLoadResult{
		ReadCloser: body,
	}
```